eduiddashboard/views/nins.py

```diff
@@ -523,13 +523,12 @@ def get_template_context(self):
 
         settings = self.request.registry.settings
         enable_mm = settings.get('enable_mm_verification')
-        session_user = get_session_user(self.request, raise_on_not_logged_in = False)
 
         context.update({
-            'nins': session_user.nins.to_list_of_dicts(),
-            'not_verified_nins': get_not_verified_nins_list(self.request, session_user),
+            'nins': self.user.nins.to_list_of_dicts(),
+            'not_verified_nins': get_not_verified_nins_list(self.request, self.user),
             'active_nin': self.get_active_nin(),
-            'has_mobile': has_confirmed_mobile(session_user),
+            'has_mobile': has_confirmed_mobile(self.user),
```

Comment:
oops, the  `add_by_mobile_success` method updates the db with a user object retrieved with get_session_user, so the user in the context may be out of date.

So I can aither make that method use the user in the context, or make the get_template_context  get the user with get_session_user. Opinions?